universe/supplycommit/multi_sm_manager.go

```diff
@@ -377,6 +377,25 @@ func (m *MultiStateMachineManager) FetchCommitment(ctx context.Context,
 	}), nil
 }
 
+// FetchSupplyLeavesByHeight returns the set of supply leaves for the given
+// asset specifier within the specified height range.
+func (m *MultiStateMachineManager) FetchSupplyLeavesByHeight(
```

Comment:
If we're exposing this for the RPC server, then can't we just give it a slim interface based on `m.cfg.TreeView`? Nothing about this method actually relates to the state machine themselves, we're only exposing a new read method.

Reply:
I mean giving the RPC server `m.cfg.TreeView` directly, or just a single method interface. As there's nothing here that actually uses the target struct (`m`) beyond the cfg access.